File: builder/java/BuilderTest.java
public class BuilderTest {
    public static void main(String[] args) {
        ResourcePoolConfig config = new ResourcePoolConfig.Builder()
                .name("jdbc-mysql.properties")
                .maxTotal(30)
                .maxIdle(30)
                .minIdle(15)
                .build();
        System.out.printf("建造者模式: %s%n", config);
    }
}

File: builder/go/builder_test.go
package builder

import "testing"

func TestBuilder(t *testing.T) {
	opts := []ResourcePoolConfigOption{
		WithMaxTotal(30),
		WithMaxIdle(30),
		WithMinIdle(15),
	}
	if conf, err := NewResourcePoolConfig("jdbc-mysql.properties", opts...); err != nil {
		t.Errorf("建造者模式：%s", err)
	} else {
		t.Logf("建造者模式: %s\n", conf)
	}
}
